### ISSUE TITLE
What's the loss to train the semantic-autoenc?

### ISSUE THREAD

**Issue by Allencheng97** [phizaz/diffae#32]
Dear diffae team,

Thank you for your great work. I have one question about some implementation detail.
I'm a little lost in the loss of training ae. 
From my understanding , the ae input is a image and output is a vector contains semantic information.
I see the L1 loss in the templates
![image](https://user-images.githubusercontent.com/47037416/191035787-a0c61765-a360-4111-8e9c-fae97712771c.png)
So what's the two item to calculate the loss ? Maybe one is the z_sem produced by encoder, what is  the other item?
Or this does not have any loss, just train with a ddim and use the loss of ddim to do backpropagation?

**Comment by phizaz** [comment]
You have multiple questions. It's not easy for me to answer to all of them. You may reread the paper (maybe after watching the presentation video).

My suggestion is you need a solid understanding of DDPM to not easily get confused with terms used in the paper. 
Don't get distracted by L1 or L2 losses in the code. They are not important for understanding.

Yes, `z_sem` is produced by the encoder. 
Yes, you just need to train DDIM and do backpropagation.

**Comment by Allencheng97** [comment]
> You have multiple questions. It's not easy for me to answer to all of them. You may reread the paper (maybe after watching the presentation video).
> 
> My suggestion is you need a solid understanding of DDPM to not easily get confused with terms used in the paper. Don't get distracted by L1 or L2 losses in the code. They are not important for understanding.
> 
> Yes, `z_sem` is produced by the encoder. Yes, you just need to train DDIM and do backpropagation.

Hi, Thank you for your reply. I checked other issues and find the answer. For me training such an encoder without any constraint but also can extract meaningful semantic information seems magic, I thought there's a certain loss to forcing the extracted feature to bind with semantics in the beginning. But thank you anyway.
